awx/main/access.py

```diff
@@ -2764,5 +2765,50 @@ def can_delete(self, obj):
         return False
 
 
+class WorkflowApprovalAccess(BaseAccess):
+    '''
+    I can approve workflows when:
+     - I'm authenticated
+    I can create when:
+     - I'm a superuser:
+    '''
+
+    model = WorkflowApproval
+    prefetch_related = ('created_by', 'modified_by',)
+
+    def can_read(self, obj):
+        return True
+
+    def can_use(self, obj):
+        return True
+
+    def filtered_queryset(self):
+        return self.model.objects.all()
+
+    def can_start(self, obj, validate_license=True):
+        return False
+
+
+class WorkflowApprovalTemplateAccess(BaseAccess):
+    '''
+    I can approve workflows when:
+     - I'm authenticated
+    I can create when:
+     - I'm a superuser:
+    '''
+
+    model = WorkflowApprovalTemplate
+    prefetch_related = ('created_by', 'modified_by',)
+
+    def can_read(self, obj):
+        return True
+
+    def can_use(self, obj):
+        return True
+
+    def filtered_queryset(self):
+        return self.model.objects.all()
```

Comment:
Spitballing from the conversation in the issue

https://github.com/ansible/awx/issues/1206#issuecomment-509748862

What you want would probably look vaguely like:

```python
    def filtered_queryset(self):
        return self.model.filter(workflowjobtemplatenodes__workflow_job_template=WorkflowJobTemplate.accessible_pk_qs(self.user, 'read_role'))
```